beacon_node/operation_pool/src/sync_aggregate_id.rs

```diff
@@ -0,0 +1,32 @@
+use serde_derive::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use types::{ChainSpec, Domain, EthSpec, Fork, Hash256, Slot};
+
+/// Used to key `SyncAggregate`s in the `naive_sync_aggregation_pool`.
+#[derive(
+    PartialEq, Eq, Clone, Hash, Debug, PartialOrd, Ord, Encode, Decode, Serialize, Deserialize,
+)]
+pub struct SyncAggregateId {
+    pub slot: Slot,
+    pub beacon_block_root: Hash256,
+    pub domain: Hash256,
```

Comment:
I don't think the `domain` is necessary here. I don't think we need it for attestations either (the topic of https://github.com/sigp/lighthouse/issues/1255), but we don't need to address that right now.

The reason I think we don't need to store the domain is that it can be determined from the slot. I.e. the slot uniquely determines the domain, and there's never a case where a node will want to collect different sync contributions for the same slot with different domains. The processing of sync aggregates at the consensus layer requires the domain to be fixed (based on the slot).